### PR TITLE
Webpack config

### PR DESCRIPTION
**Pull request by athensbird**
- add loaders to dynamically transform multiple file types
- configure webpack to work with React Router
- change default public path to enable webpack dev server auto-reloading
#119

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -5,8 +5,12 @@ import Auth from './components/Auth';
 import HeadBar from './components/HeadBar';
 import ProjectGrid from './components/ProjectGrid';
 
+declare const __FRONTEND__: string;
+const fn = __FRONTEND__;
+
 class App extends React.Component {
   public render() {
+    console.log(fn);
```
Comment (vrann) [inline frontend/src/App.tsx]: please remove debug

File: frontend/src/components/ProjectCard.tsx [hunk 2]
```diff
@@ -76,8 +76,8 @@ const styles = (theme:any) => ({
     'margin-left': 'auto',
   },
   progress: {
-    color: '#48BF61',
-    colorSecondary: '#FF0000',
+    // color: '#48BF61',
```
Comment (vrann) [inline frontend/src/components/ProjectCard.tsx]: please remove debug

File: frontend/src/components/buttons/LoginButton.tsx [hunk 3]
```diff
@@ -4,16 +4,25 @@ import Typography from '@material-ui/core/Typography';
...
+  authWrapper: {
+    display: 'flex',
+    'align-items': 'center',
+  },
   logoutButton: {
-    [theme.breakpoints.down('sm')]: {
-      position: 'relative' as 'relative',
-      bottom: '1rem',
-    },
+    // [theme.breakpoints.down('xs')]: {
```
Comment (vrann) [inline frontend/src/components/buttons/LoginButton.tsx]: please remove debug

File: frontend/src/components/HeadBar.tsx [hunk 4]
```diff
@@ -41,13 +45,12 @@ const HeadBar = (props: any) => {
         <Logo />
         <AddProject className={classes.addButton} />
         <Login
-          clientId="668e0b6c450cc783f267"
+          clientId={gitId}
```
Comment (vrann) [inline frontend/src/components/HeadBar.tsx]: please replace the hardcoded instance of GIT_ID in frontend/src/components/WithAuth.tsx after mergig with latest staging

File: frontend/src/App.tsx (hunk 1 above)
Comment (athensbird) [reply frontend/src/App.tsx]: removed

File: frontend/src/components/ProjectCard.tsx (hunk 2 above)
Comment (athensbird) [reply frontend/src/components/ProjectCard.tsx]: removed

File: frontend/src/components/buttons/LoginButton.tsx (hunk 3 above)
Comment (athensbird) [reply frontend/src/components/buttons/LoginButton.tsx]: removed

File: frontend/src/components/HeadBar.tsx (hunk 4 above)
Comment (athensbird) [reply frontend/src/components/HeadBar.tsx]: Will do!

Comment (vrann) [reply frontend/src/components/HeadBar.tsx]: should be part of this PR

Comment (athensbird) [reply frontend/src/components/HeadBar.tsx]: I believe the GIT_ID is also passed from local environment in `withAuth.tsx` - should I take a look at the stating branch?

Comment (vrann) [reply frontend/src/components/HeadBar.tsx]: please merge with the latest sprint-dev and fix if needed. Now it is in different line:
https://github.com/magento-engcom/community-portal/blob/sprint-dev/frontend/src/components/WithAuth.tsx#L66

**Pull request merged**